partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/DiadicOperator.kt

```diff
@@ -506,11 +485,23 @@ internal abstract class DiadicOperator(
 
         // Decimal
         fillTable(PType.DECIMAL, PType.DECIMAL) { lhs, rhs -> getDecimalInstance(lhs, rhs) }
-        fillTable(PType.UNKNOWN, PType.DECIMAL) { lhs, rhs -> getDecimalInstance(rhs, rhs) }
-        fillTable(PType.DECIMAL, PType.UNKNOWN) { lhs, rhs -> getDecimalInstance(lhs, lhs) }
     }
 
     private fun fillUnknownTable() {
+        val family = SqlTypeFamily.NUMBER.members +
+            SqlTypeFamily.TIME.members +
+            SqlTypeFamily.TIMESTAMP.members +
+            SqlTypeFamily.BOOLEAN.members +
+            SqlTypeFamily.DATE.members +
+            PType.INTERVAL_YM +
+            PType.INTERVAL_DT +
+            PType.BLOB
+
+        family.forEach {
+            fillTable(PType.UNKNOWN, it) { lhs, rhs -> getUnknownInstance(lhs, rhs) }
+            fillTable(it, PType.UNKNOWN,) { lhs, rhs -> getUnknownInstance(lhs, rhs) }
+        }
```

Comment:
This is missing some types. Rather than calling iterating through all of the sql type families, you should probably just iterate through the `PType` codes -- https://github.com/partiql/partiql-lang-kotlin/blob/dixiao/intervalbugfixes/partiql-spi/src/main/java/org/partiql/spi/types/PType.java#L170. And filter out the `UNKNOWN`.